src/geofabrics/lidar_fetch.py

```diff
@@ -0,0 +1,155 @@
+        self.verbose = verbose
+        
+        self._dataset_prefixes = None
+        
+    def run(self):
+        """ Querey for  LiDAR data within a catchment and download any that hasn't already been downloaded """
+        self._dataset_prefixes = []
+        
+        json_response = self.query_for_datasets_inside_catchment()
+        
+        ot_endpoint_url = urllib.parse.urlunparse((self.SCHEME, self.NETLOC_DATA, "", "", "", ""))
+        client = boto3.client('s3', endpoint_url=ot_endpoint_url, 
+                              config=botocore.config.Config(signature_version=botocore.UNSIGNED))
+        
+        # cycle through each dataset within a region
+        for i in range(len(json_response['Datasets'])):
+            dataset_prefix = json_response['Datasets'][i]['Dataset']['alternateName']
+            self._dataset_prefixes.append(dataset_prefix)
+            
+            tile_info = self._get_dataset_tile_info(client, dataset_prefix)
+            
+            # check download size limit is not exceeded
+            lidar_size_bytes = self._calculate_dataset_download_size(client, dataset_prefix, tile_info)
+            
+            assert lidar_size_bytes/1000/1000/1000 < self.download_limit_gbytes, "The size of the LiDAR to be " \
+                + "downloaded is greater than the specified download limit of " + str(self.download_limit_gbytes)
+            
+            # check for tiles and download as needed
+            self._download_tiles_in_catchment(client, dataset_prefix, tile_info)
+            
+        
+    def query_for_datasets_inside_catchment(self):
+        """ Function to check for data in search region using hte otCatalogue API
+        https://portal.opentopography.org/apidocs/#/Public/getOtCatalog """
+        
+        api_queary = {
+            "productFormat": "PointCloud",
+            "minx": self.catchment_geometry.catchment.geometry.to_crs(self.CRS).bounds['minx'].min(),
+            "miny": self.catchment_geometry.catchment.geometry.to_crs(self.CRS).bounds['miny'].min(),
+            "maxx": self.catchment_geometry.catchment.geometry.to_crs(self.CRS).bounds['maxx'].max(),
+            "maxy": self.catchment_geometry.catchment.geometry.to_crs(self.CRS).bounds['maxy'].max(),
+            "detail": False,
+            "outputFormat": "json",
+            "inlcude_federated": True
+        }
+        
+        data_url = urllib.parse.urlunparse((self.SCHEME, self.NETLOC_API, self.PATH_API, "", "", ""))
+        
+        response = requests.get(data_url, params=api_queary, stream=True)
+        response.raise_for_status()
+        return response.json()
+        
+    def _get_dataset_tile_info(self, client, dataset_prefix):
+        """ Check for the tile index shapefile and download as needed, then load in 
+        and trim to the catchment to determine which data tiles to download. """ 
+        
+        
+        file_prefix = dataset_prefix + "/" + dataset_prefix + "_TileIndex.zip"
```

Comment:
You could consider this, but let me know what you think
```suggestion
        file_prefix = f"{dataset_prefix}/{dataset_prefix}_TileIndex.zip"
```

Reply:
I like it! I should and will make better use of string formatting in that way!